In [17]:
import pandas as pd
import numpy as np

# Import Data

In [18]:
df_netflix = pd.read_csv(r"C:\Users\User\Desktop\ai\netflix_titles.csv")

In [29]:
df_netflix.head(13)


show_id     type                                title  \
0       s1    Movie                 Dick Johnson Is Dead   
1       s2  TV Show                        Blood & Water   
2       s3  TV Show                            Ganglands   
3       s4  TV Show                Jailbirds New Orleans   
4       s5  TV Show                         Kota Factory   
5       s6  TV Show                        Midnight Mass   
6       s7    Movie     My Little Pony: A New Generation   
7       s8    Movie                              Sankofa   
8       s9  TV Show        The Great British Baking Show   
9      s10    Movie                         The Starling   
10     s11  TV Show  Vendetta: Truth, Lies and The Mafia   
11     s12  TV Show                     Bangkok Breaking   
12     s13    Movie                         Je Suis Karl   

                         director  \
0                 Kirsten Johnson   
1                             NaN   
2                 Julien Leclercq   
3                             NaN   
4                             NaN   
5                   Mike Flanagan   
6   Robert Cullen, José Luis Ucha   
7                    Haile Gerima   
8                 Andy Devonshire   
9                  Theodore Melfi   
10                            NaN   
11              Kongkiat Komesiri   
12            Christian Schwochow   

                                                 cast  \
0                                                 NaN   
1   Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   
2   Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...   
3                                                 NaN   
4   Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...   
5   Kate Siegel, Zach Gilford, Hamish Linklater, H...   
6   Vanessa Hudgens, Kimiko Glenn, James Marsden, ...   
7   Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
8   Mel Giedroyc, Sue Perkins, Mary Berry, Paul Ho...   
9   Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   
10                                                NaN   
11  Sukollawat Kanarot, Sushar Manaying, Pavarit M...   
12  Luna Wedler, Jannis Niewöhner, Milan Peschel, ...   

                                              country          date_added  \
0                                       United States  September 25, 2021   
1                                        South Africa  September 24, 2021   
2                                                 NaN  September 24, 2021   
3                                                 NaN  September 24, 2021   
4                                               India  September 24, 2021   
5                                                 NaN  September 24, 2021   
6                                                 NaN  September 24, 2021   
7   United States, Ghana, Burkina Faso, United Kin...  September 24, 2021   
8                                      United Kingdom  September 24, 2021   
9                                       United States  September 24, 2021   
10                                                NaN  September 24, 2021   
11                                                NaN  September 23, 2021   
12                            Germany, Czech Republic  September 23, 2021   

    release_year rating   duration  \
0           2020  PG-13     90 min   
1           2021  TV-MA  2 Seasons   
2           2021  TV-MA   1 Season   
3           2021  TV-MA   1 Season   
4           2021  TV-MA  2 Seasons   
5           2021  TV-MA   1 Season   
6           2021     PG     91 min   
7           1993  TV-MA    125 min   
8           2021  TV-14  9 Seasons   
9           2021  PG-13    104 min   
10          2021  TV-MA   1 Season   
11          2021  TV-MA   1 Season   
12          2021  TV-MA    127 min   

                                            listed_in  \
0                                       Documentaries   
1     International TV Shows, TV Dramas, TV Mysteries   
2   Crime TV Shows, International TV Shows, TV Act...   
3   

**Tail**

In [11]:
df_netflix.tail(10)

show_id     type               title                 director  \
8797   s8798  TV Show           Zak Storm                      NaN   
8798   s8799    Movie            Zed Plus  Chandra Prakash Dwivedi   
8799   s8800    Movie               Zenda           Avadhoot Gupte   
8800   s8801  TV Show  Zindagi Gulzar Hai                      NaN   
8801   s8802    Movie             Zinzana          Majid Al Ansari   
8802   s8803    Movie              Zodiac            David Fincher   
8803   s8804  TV Show         Zombie Dumb                      NaN   
8804   s8805    Movie          Zombieland          Ruben Fleischer   
8805   s8806    Movie                Zoom             Peter Hewitt   
8806   s8807    Movie              Zubaan              Mozez Singh   

                                                   cast  \
8797  Michael Johnston, Jessica Gee-George, Christin...   
8798  Adil Hussain, Mona Singh, K.K. Raina, Sanjay M...   
8799  Santosh Juvekar, Siddharth Chandekar, Sachit P...   
8800  Sanam Saeed, Fawad Khan, Ayesha Omer, Mehreen ...   
8801  Ali Suliman, Saleh Bakri, Yasa, Ali Al-Jabri, ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...   
8803                                                NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...   

                                            country          date_added  \
8797  United States, France, South Korea, Indonesia  September 13, 2018   
8798                                          India   December 31, 2019   
8799                                          India   February 15, 2018   
8800                                       Pakistan   December 15, 2016   
8801                   United Arab Emirates, Jordan       March 9, 2016   
8802                                  United States   November 20, 2019   
8803                                            NaN        July 1, 2019   
8804                                  United States    November 1, 2019   
8805                                  United States    January 11, 2020   
8806                                          India       March 2, 2019   

      release_year rating   duration  \
8797          2016  TV-Y7  3 Seasons   
8798          2014  TV-MA    131 min   
8799          2009  TV-14    120 min   
8800          2012  TV-PG   1 Season   
8801          2015  TV-MA     96 min   
8802          2007      R    158 min   
8803          2018  TV-Y7  2 Seasons   
8804          2009      R     88 min   
8805          2006     PG     88 min   
8806          2015  TV-14    111 min   

                                              listed_in  \
8797                                           Kids' TV   
8798             Comedies, Dramas, International Movies   
8799                       Dramas, International Movies   
8800  International TV Shows, Romantic TV Shows, TV ...   
8801            Dramas, International Movies, Thrillers   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description  
8797  Teen surfer Zak Storm is mysteriously transpor...  
8798  A philandering small-town mechanic's political...  
8799  A change in the leadership of a political part...  
8800  Strong-willed, middle-class Kashaf and carefre...  
8801  Recovering alcoholic Talal wakes up inside a s...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...

In [13]:
df_netflix.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


In [14]:
df_netflix.shape

(8807, 12)

In [16]:
df_netflix.describe()

release_year
count   8807.000000
mean    2014.180198
std        8.819312
min     1925.000000
25%     2013.000000
50%     2017.000000
75%     2019.000000
max     2021.000000

# 1. Rows and columns in the data set

In [20]:
df_netflix.shape

(8807, 12)

**There are 8807 rows and 12 columns in the dataset**

# 2. Datatypes contained in the data set

In [21]:
df_netflix.dtypes

show_id         object
type            object
title           object
director        object
cast            object
country         object
date_added      object
release_year     int64
rating          object
duration        object
listed_in       object
description     object
dtype: object

****The datatypes are object and int****

# 3. Is the dataset complete?

In [22]:
df_netflix.isnull().sum()

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

**The data set is not complete. There are some null values present**

# 4. The number of Movies and TV Shows among the last 20 rows of the dataset.

In [23]:
df_netflix.tail(20)['type'].value_counts()

type
Movie      15
TV Show     5
Name: count, dtype: int64

**There are 15 movies and 5 TV shows in the last 20 rows**

# 5. Give the number of all Movies originating from South Africa

In [47]:
sa_movies = df_netflix[(df_netflix['type'] == 'Movie') & (df_netflix['country'] == 'South Africa')]
count = len(sa_movies)

print(f"Number of movies from South Africa: {count}")

Number of movies from South Africa: 24


# 6. Description of the movie in the 12th row

In [27]:
df_netflix.loc[11,'description']

'Struggling to earn a living in Bangkok, a man joins an emergency rescue service and realizes he must unravel a citywide conspiracy.'

**'Struggling to earn a living in Bangkok, a man joins an emergency rescue service and realizes he must unravel a citywide conspiracy.**

# 7.South African movies released in 2021

In [39]:
south_africa_movies = df_netflix[(df_netflix['type'] == 'Movie') & (df_netflix['country'] == 'South Africa')]
south_africa_2021 = south_africa_movies[df_netflix['release_year'] == 2021]
print("South African movies released in 2021:")
print(south_africa_2021[['title', 'release_year']], "\n")

South African movies released in 2021:
                             title  release_year
293                           Slay          2021
764   Trippin' with the Kandasamys          2021
881                 I Am All Girls          2021
1172       Loyiso Gola: Unlearning          2021 



C:\Users\User\AppData\Local\Temp\ipykernel_15288\3454812762.py:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  south_africa_2021 = south_africa_movies[df_netflix['release_year'] == 2021]


# 8.List the entire cast of all movies released in 2021.

In [46]:
movies_2021 = df_netflix[(df_netflix['type'] == 'Movie') & (df_netflix['release_year'] == 2021)]

print("Cast of all movies released in 2021:\n")
for index, row in movies_2021.dropna(subset=['cast']).iterrows():
    print(f"{row['title']}: {row['cast']}")

Cast of all movies released in 2021:

My Little Pony: A New Generation: Vanessa Hudgens, Kimiko Glenn, James Marsden, Sofia Carson, Liza Koshy, Ken Jeong, Elizabeth Perkins, Jane Krakowski, Michael McKean, Phil LaMarr
The Starling: Melissa McCarthy, Chris O'Dowd, Kevin Kline, Timothy Olyphant, Daveed Diggs, Skyler Gisondo, Laura Harrier, Rosalind Chao, Kimberly Quinn, Loretta Devine, Ravi Kapoor
Je Suis Karl: Luna Wedler, Jannis Niewöhner, Milan Peschel, Edin Hasanović, Anna Fialová, Marlon Boess, Victor Boccard, Fleur Geffrier, Aziz Dyab, Mélanie Fouché, Elizaveta Maximová
Confessions of an Invisible Girl: Klara Castanho, Lucca Picon, Júlia Gomes, Marcus Bessa, Kiria Malheiros, Fernanda Concon, Gabriel Lima, Caio Cabral, Leonardo Cidade, Jade Cardozo
Intrusion: Freida Pinto, Logan Marshall-Green, Robert John Burke, Megan Elisabeth Kelly, Sarah Minnich, Hayes Hargrove, Mark Sivertsen, Brandon Fierro, Antonio Valles, Clint Obenchain
Go! Go! Cory Carson: Chrissy Takes the Wheel: Maisie B